ttnn/cpp/ttnn/tensor/tensor_impl.cpp

```diff
@@ -556,20 +557,32 @@ std::string to_string<bfloat4_b>(const Tensor& tensor, std::optional<DataType> o
 template <typename T>
 Tensor to_host_helper(const Tensor& tensor, bool blocking = true, uint8_t cq_id = ttnn::DefaultQueueId) {
     TT_ASSERT(tensor.is_allocated(), "Buffer must be allocated on device!");
-    auto device_buffer = tensor.device_buffer();
-    auto device = tensor.device();
+
+    Tensor input_tensor;
+    // Currently no direct sharded read support in BLACKHOLE due to alignment issue
+    if(tensor.is_sharded ()  and (tensor.device()->arch() == tt::ARCH::BLACKHOLE)) {
+        MemoryConfig dram_memory_config = tt::tt_metal::MemoryConfig{.memory_layout=tt::tt_metal::TensorMemoryLayout::INTERLEAVED,.buffer_type=tt::tt_metal::BufferType::DRAM};
```

Comment:
Why do we spill out the output to DRAM?

Reply:
oh true, we can use L1, if it fits sharded it should fit in L1